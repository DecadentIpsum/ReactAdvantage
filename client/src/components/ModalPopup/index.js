import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Dialog } from 'primereact/components/dialog/Dialog';

import './index.css';

const modalRoot = document.getElementById('modal-root');

export default class ModalPopup extends Component {
    constructor(props) {
        super(props);
        this.el = document.createElement('div');
    }

    componentDidMount() {
        modalRoot && modalRoot.appendChild(this.el);
    }

    componentWillUnmount() {
        modalRoot && modalRoot.removeChild(this.el);
    }

    render() {
        const { inPortal, ...props } = this.props;

        if (!inPortal) {
            return (
                <div className="dialog-wrapper">
                    <Dialog {...props}>
                        {this.props.children}
                    </Dialog>
                </div>
            );
        }

        return (
            ReactDOM.createPortal(
                <div className="dialog-wrapper">
                    <Dialog {...props}>
                        {this.props.children}
                    </Dialog>
                </div>,
                this.el
            )
        );
    }
}
