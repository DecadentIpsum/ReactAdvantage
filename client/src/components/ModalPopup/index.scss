@import "global-styles/common.scss";

.dialog-wrapper {
    font-size: 1rem;
}