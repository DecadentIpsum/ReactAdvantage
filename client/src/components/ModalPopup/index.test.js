import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import ModalPopup from './index';

it('renders correctly', () => {
    const component = shallow(<ModalPopup onHide={() => {}} />);

    expect(component).toMatchSnapshot();
});