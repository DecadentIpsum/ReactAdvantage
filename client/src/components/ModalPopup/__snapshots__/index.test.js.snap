// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalPopup
    onHide={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Dialog
        appendTo={null}
        autoAlign={true}
        baseZIndex={0}
        breakpoint={640}
        className={null}
        closable={true}
        closeOnEscape={true}
        contentStyle={null}
        dismissableMask={false}
        draggable={true}
        footer={null}
        header={null}
        height="auto"
        id={null}
        minHeight={150}
        minWidth={150}
        minX={0}
        minY={0}
        modal={false}
        onHide={[Function]}
        onShow={null}
        positionLeft={-1}
        positionTop={-1}
        resizable={true}
        responsive={true}
        rtl={false}
        showHeader={true}
        style={null}
        visible={false}
        width="auto"
/>,
      "className": "dialog-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appendTo": null,
        "autoAlign": true,
        "baseZIndex": 0,
        "breakpoint": 640,
        "children": undefined,
        "className": null,
        "closable": true,
        "closeOnEscape": true,
        "contentStyle": null,
        "dismissableMask": false,
        "draggable": true,
        "footer": null,
        "header": null,
        "height": "auto",
        "id": null,
        "minHeight": 150,
        "minWidth": 150,
        "minX": 0,
        "minY": 0,
        "modal": false,
        "onHide": [Function],
        "onShow": null,
        "positionLeft": -1,
        "positionTop": -1,
        "resizable": true,
        "responsive": true,
        "rtl": false,
        "showHeader": true,
        "style": null,
        "visible": false,
        "width": "auto",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Dialog
          appendTo={null}
          autoAlign={true}
          baseZIndex={0}
          breakpoint={640}
          className={null}
          closable={true}
          closeOnEscape={true}
          contentStyle={null}
          dismissableMask={false}
          draggable={true}
          footer={null}
          header={null}
          height="auto"
          id={null}
          minHeight={150}
          minWidth={150}
          minX={0}
          minY={0}
          modal={false}
          onHide={[Function]}
          onShow={null}
          positionLeft={-1}
          positionTop={-1}
          resizable={true}
          responsive={true}
          rtl={false}
          showHeader={true}
          style={null}
          visible={false}
          width="auto"
/>,
        "className": "dialog-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appendTo": null,
          "autoAlign": true,
          "baseZIndex": 0,
          "breakpoint": 640,
          "children": undefined,
          "className": null,
          "closable": true,
          "closeOnEscape": true,
          "contentStyle": null,
          "dismissableMask": false,
          "draggable": true,
          "footer": null,
          "header": null,
          "height": "auto",
          "id": null,
          "minHeight": 150,
          "minWidth": 150,
          "minX": 0,
          "minY": 0,
          "modal": false,
          "onHide": [Function],
          "onShow": null,
          "positionLeft": -1,
          "positionTop": -1,
          "resizable": true,
          "responsive": true,
          "rtl": false,
          "showHeader": true,
          "style": null,
          "visible": false,
          "width": "auto",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
